### PR TITLE
Fix handling of user labels on custom resources

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Bugfix

### Description

We do a validation of the labels set on the custom resources. They should not contain any `strimzi.io` labels. However, for legacy reasons from when we used ConfigMaps we allowed the `strimzi.io/kind` label which defined which resource should be created. This label is not needed anymore. This PR cleans the check and the error message which is raised when `strimzi.io` labels are used.

### Checklist

- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally

### CODE REVIEW

**Comment by scholzj** [comment]
@strimzi-ci run tests

**Comment by tombentley** [comment]
Should the upgrade instructions be modified to tell users to remove `strimzi.io/kind` prior to upgrading?

**Comment by scholzj** [comment]
@tombentley Do you actually expect anyone to use that label?

File: operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java [hunk 1]
```diff
@@ -82,10 +82,8 @@ public static String name(HasMetadata resource) {
     public static Labels userLabels(Map<String, String> userLabels) {
         if (userLabels != null) {
             for (String key : userLabels.keySet()) {
-                if (key.startsWith(STRIMZI_DOMAIN)
-                        && !key.equals(STRIMZI_KIND_LABEL)) {
-                    throw new IllegalArgumentException("User labels includes a Strimzi label that is not "
-                            + STRIMZI_KIND_LABEL + ": " + key);
+                if (key.startsWith(STRIMZI_DOMAIN)) {
+                    throw new IllegalArgumentException("Labels starting with " + STRIMZI_DOMAIN + " are not allowed at Custom Resources.");
```
Comment (tombentley) [inline operator-common/src/main/java/io/strimzi/operator/common/model/Labels.java]:
```suggestion
                    throw new IllegalArgumentException("Labels starting with " + STRIMZI_DOMAIN + " are not allowed in Custom Resources, such labels should be removed.");
```

**Comment by scholzj** [comment]
@strimzi-ci run tests

**Comment by scholzj** [comment]
@strimzi-ci run tests testcase=KafkaST#testLoadBalancerTls,RecoveryST#RecoveryST,AllNamespaceST#testKafkaInDifferentNsThanClusterOperator,ConnectS2IST#testDeployS2IWithMongoDBPlugin,SecurityST#testAutoRenewCaCertsTriggeredByAnno,UserST#testUserWithNameMoreThan64Chars,UserST#testBigAmountOfTlsUsers

**Comment by scholzj** [comment]
@strimzi-ci run tests testcase=ConnectS2IST#testDeployS2IWithMongoDBPlugin,SecurityST#testAutoRenewCaCertsTriggeredByAnno,UserST

**Comment by scholzj** [comment]
@strimzi-ci <https://github.com/strimzi-ci> run tests
testcase=ConnectS2IST#testDeployS2IWithMongoDBPlugin,SecurityST#testAutoRenewCaCertsTriggeredByAnno,UserST

**Comment by scholzj** [comment]
@strimzi-ci run tests testcase=ConnectS2IST#testDeployS2IWithMongoDBPlugin

**Pull request merged**